plugins/helpers.js

```diff
@@ -1,5 +1,28 @@
+const asyncHooks = require('async_hooks');
+const state = new Map();
+
+const breakLoop = function(name, fn) {
+    return function(...args) {
+        const asyncId = asyncHooks.executionAsyncId();
+        const events = state.get(asyncId) || [];
+        if (!events.includes(name)) {
+            state.set(asyncId, events.concat(name));
+            fn(...args);
+        }
+    };
+};
+
+const hook = asyncHooks.createHook({
+    init: (aid, type, tid) => {
+        if (state.has(tid)) state.set(aid, state.get(tid));
+    },
+    destroy: (aid) => state.delete(aid)
+});
```

Comment:
I had to use an AsyncHook to trace down any problematic asynchronous call.
In short, it prevents calling `src.emit('event')` while calling `dst.emit('event')` which causes an infinite loop